var/spack/repos/builtin/packages/cppad/package.py

```diff
@@ -0,0 +1,45 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Cppad(CMakePackage):
+    """A Package for Differentiation of C++ Algorithms."""
+
+    homepage = "https://www.coin-or.org/CppAD/"
+    url      = "http://www.coin-or.org/download/source/CppAD/cppad-20170114.gpl.tgz"
+
+    version('20170114.gpl', '565a534dc813fa1289764222cd8c11ea')
```

Comment:
Sorry, I don't know what the difference for the licence between the two versions is.  I tested without that part in the version string and found that it is unfortunately necessary (my commit is based off of `d3d87ea`). It appears to me that spack is not detecting that the file's full suffix is `.gpl.tgz`?

```bash
$ spack install cppad
==> Installing cppad
==> cmake is already installed in /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/cmake-3.6.1-dbk37r7pllmngnjzo4umf4hbszv2j7i7
==> Fetching http://www.coin-or.org/download/source/CppAD/cppad-20170114.tgz
######################################################################## 100.0%
curl: (22) The requested URL returned error: 404 Not Found
==> Fetching from http://www.coin-or.org/download/source/CppAD/cppad-20170114.tgz failed.
==> Error: FetchError: All fetchers failed for cppad-20170114-jm5ietsxi6c4zz53o3e3q4fhwe7udkvz
/Users/jp/Programs/GitHub/Spack/spack/lib/spack/spack/package.py:965, in do_fetch:
     938      def do_fetch(self, mirror_only=False):
     939          """
     940          Creates a stage directory and downloads the tarball for this package.
     941          Working directory will be set to the stage directory.
     942          """
     943          if not self.spec.concrete:
     944              raise ValueError("Can only fetch concrete packages.")
     945
     946          start_time = time.time()
     947          if spack.do_checksum and self.version not in self.versions:
     948              tty.warn("There is no checksum on file to fetch %s safely." %
     949                       self.spec.format('$_$@'))
     950
     951              # Ask the user whether to skip the checksum if we're
     952              # interactive, but just fail if non-interactive.
     953              ck_msg = "Add a checksum or use --no-checksum to skip this check."
     954              ignore_checksum = False
     955              if sys.stdout.isatty():
     956                  ignore_checksum = tty.get_yes_or_no("  Fetch anyway?",
     957                                                      default=False)
     958                  if ignore_checksum:
     959                      tty.msg("Fetching with no checksum.", ck_msg)
     960
     961              if not ignore_checksum:
     962                  raise FetchError("Will not fetch %s" %
     963                                   self.spec.format('$_$@'), ck_msg)
     964
  >> 965          self.stage.fetch(mirror_only)
     966
     967          self._fetch_time = time.time() - start_time
     968
     969          if spack.do_checksum and self.version in self.versions:
     970              self.stage.check()
     971
     972          self.stage.cache_local()
```